content/tutorials/reactjs-realtime-commenting.textile

```diff
@@ -0,0 +1,344 @@
+  // Prevent the default behaviour of form submit
+  e.preventDefault();
+
+  // Get the value of the comment box
+  // and make sure it not some empty strings
+  const comment = e.target.elements.comment.value.trim();
+  const name = e.target.elements.name.value.trim();
+
+  // Make sure name and comment boxes are filled
+  if (name && comment) {
+    const commentObject = { name, comment };
+
+    // Publish comment
+    const channel = Ably.channels.get('comments');
+    channel.publish('add_comment', commentObject);
+
+    // Clear input fields
+    e.target.elements.comment.value = '';
+    e.target.elements.name.value = '';
+  }
+}
+```
+
+We first prevent the default form submission behaviour (that is, reloading the page). We then get the form inputs entered (name and comment) and make sure they are not just some empty strings. If the inputs are properly filled, we publish the comment just made to a @comments@  channel with an @add_comment@ event. Lastly, we clear the form fields.
+
+h2. Step 10 - Create Comment Component
+
+Within the @components@ folder, create a new file named @Comment.js@ and add the code below into it:
+
+```[javascript]
+import React, { Component } from 'react';
+
+class Comment extends Component {
+  render() {
+    return (
+      <article className="media">
+        <figure className="media-left">
+          <p className="image is-64x64">
+           <img src="https://bulma.io/images/placeholders/128x128.png" alt="Avatar" />
+          </p>
+        </figure>
+        <div className="media-content">
+          <div className="content">
+            <p>
+              <strong>{this.props.comment.name}</strong>
+              <br />
+              {this.props.comment.comment}
+            </p>
+          </div>
+        </div>
+      </article>
+    );
+  }
+}
+
+export default Comment;
+```
+
+This simply renders a single comment. It accepts the comment as props.
+
+h2. Step 11 - Create Comments Component
+
+Within the @components@ folder, create a new file named @Comments.js@ and add the code below into it:
+
+```[javascript]
+import React, { Component } from 'react';
+import Comment from './Comment';
+
+class Comments extends Component {
+  render() {
+    return (
+      <section className="section">
+        {
+          this.props.comments.map((comment, index) => {
+            return <Comment key={index} comment={comment} />
+          })
+        }
+      </section>
+    );
+  }
+}
+
+export default Comments;
+```
+
+This component accepts a @comments@ props and renders the Comment component for each of the comments available. It passes the actual comment to the Comment component as props.
+
+h2. Step 12 - Create App Component
+
+The App component will server as the parent component. Open @components/App.js@ and replace it with the code below:
+
+```[javascript]
+import React, { Component } from 'react';
+import CommentBox from './CommentBox';
+import Comments from './Comments';
+import Ably from '../ably';
```

Comment:
Oh I see it's used in a later step. So firstly this is a problem because it does result in two connections to Ably, but also because it's just not right i.e. there should be a global Ably object shared between the components, or perhaps a separate object that handles publishing & receiving events.